src/state/nftMarket/reducer.ts

```diff
@@ -123,8 +124,21 @@ export const fetchNftsFromCollections = createAsyncThunk(
         })
       }
 
-      // TODO: revisit this for other collecitons
-      return []
+      return Object.keys(nfts.data).map((id) => {
+        const apiMetadata = nfts.data[id]
+        const marketData = nftsMarket[id] || {}
```

Comment:
this should probably `nftMarket.find((nft) => nft.tokenId === id)`. Cause right now it relies on the fact that array returned from subgraph is ordered from 0 to 10000, which is not the case if some are not on sale and/or subgraph orders them not by tokenId

Reply:
Smart